ingenialink/dictionary.py

```diff
@@ -314,6 +332,56 @@ def get_safety_tpdo(self, uid: str) -> DictionarySafetyPDO:
             return self.safety_tpdos[uid]
         raise KeyError(f"Safe TPDO {uid} not exist")
 
+    def _merge_registers(self, other_dict: "Dictionary") -> None:
+        """Add the registers from another dictionary to the dictionary instance.
+
+        Args:
+            other_dict: The other dictionary instance.
+
+        """
+        for subnode, registers in other_dict._registers.items():
+            self._registers[subnode].update(registers)
+
+    def _merge_errors(self, other_dict: "Dictionary") -> None:
+        """Add the errors from another dictionary to the dictionary instance.
+
+        Args:
+            other_dict: The other dictionary instance.
+
+        """
+        self.errors.errors.update(other_dict.errors.errors)
+
+    def _set_image(self, other_dict: "Dictionary") -> None:
+        """Set the image attribute.
+
+        Choose the image from the dictionary that has one.
+
+        Args:
+            other_dict: The other dictionary instance.
+
+        """
+        core_dict = self if self._is_coco_dictionary(other_dict) else other_dict
+        self.image = core_dict.image
+
+    def _merge_attributes(self, other_dict: "Dictionary") -> None:
+        """Add the revision number, product code, firmware version and part number
+        from the other dictionary to the dictionary instance.
+
+        Args:
+            other_dict: The other dictionary instance.
+
+        """
+        if not self._is_coco_dictionary(other_dict):
+            self.product_code = other_dict.product_code
+            self.revision_number = other_dict.revision_number
+            self.firmware_version = other_dict.firmware_version
+            self.part_number = other_dict.part_number
+
+    @staticmethod
+    def _is_coco_dictionary(dictionary: "Dictionary") -> bool:
```

Comment:
Could this be a property instead of a static method?

Reply:
Done